pkg/ebpf/netlink/conntracker.go

```diff
@@ -110,12 +110,12 @@ func newConntrackerOnce(procRoot string, deleteBufferSize, maxStateSize int) (Co
 	netns := getGlobalNetNSFD(procRoot)
 
 	logger := getLogger()
-	nfct, err := ct.Open(&ct.Config{ReadTimeout: 10 * time.Millisecond, NetNS: netns, Logger: logger})
+	nfct, err := ct.Open(&ct.Config{NetNS: netns, Logger: logger})
```

Comment:
This is what was blocking us from upgrading to 1.12. 
* For Go `< 1.12` this option was being ignored by `go-conntrack` and we were blocking indefinitely when polling the netlink socket (which is actually the desired behavior).
* By leaving this `ReadTimeout: 10 * time.Millisecond` when upgrading to Go version `1.12.9` we noticed that conntrack events stopped coming through in some cases. That is because there is virtually no error handling for timeouts in the library. So if a timeout occurs (eg. if no conntrack events are written to the netlink socket within 10 milliseconds in our case) the go routine polling the socket would terminate and we would no longer receive conntrack events.

Reply:
Nice find!